lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb

```diff
@@ -35,7 +35,7 @@ def set_up_namespace_thread
       # processing will be swallowed and retried. These failures /
       # exceptions could be caused by Kubernetes API being temporarily
       # down. We assume the configuration is correct at this point.
-      while thread_current_running?
+      while true
```

Comment:
I presume the previous logic had some kind of graceful exit?  Do we have concerns with an infinite loop and graceful shutdown?  I don't know the answer but am asking the question.  Otherwise lgtm

Reply:
Yeah, the design of [thread_current_running](https://github.com/fluent/fluentd/blob/2df5ed1f7f5f2190285642b780b1fd4bc73887d8/lib/fluent/plugin_helper/thread.rb) is for graceful exit. I dug through their code a bit. It seems like an overkill for the cases we are handling here. Thus we proposed to just use the native threading.

If the watcher is healthy, we do expect it to be an infinite loop.

Reply:
We had problems with an infinite loop related to this code (not triggering the backoff). I described it in #213

Reply:
We likely should accept this PR as is and address #213 in a subsequent change

Reply:
Hold off on merging, there's actually a huge oversight on my part, I am calling a non-existent method `setup_pod_thread`, when it's supposed to be `set_up_pod_thread`, yet things are working as expected, something is either very wrong with our validation or I'm not testing what I thought I'm testing.